### PR TITLE
fix(task): parse attached run option values

### PR DESCRIPTION
**Pull request by jdx**
## Summary

- normalize attached values for short `mise run` options before clap parses task positionals
- support forms such as `-j1`, `-Cdir`, and `-oquiet`
- add regression coverage for attached short-option values before a task name

## Root cause

The run task positional accepts hyphen-prefixed values so task arguments can be forwarded. In that parsing mode, clap treated short options with attached values as positional task names. Splitting known value-taking short options into their flag and value before parsing removes the ambiguity while preserving task-side arguments.

## Testing

- `cargo test --bin mise cli::tests`
- `hk run check --safe --format json --files0-from -` scoped to `src/cli/mod.rs`
- manual end-to-end checks for `mise run -j1 hello`, `mise run -C<dir> hello`, and `mise run -oquiet hello`

Addresses https://github.com/jdx/mise/discussions/12057

*AI-assisted — Tool: Codex; model: unavailable; version: unavailable.*

---

> [!NOTE]
> **Low Risk**
> Scoped to CLI argument preprocessing for `run`; behavior change only affects previously broken attached short-option forms, with unit tests covering edge cases.
> 
> **Overview**
> Fixes **`mise run`** so short options with **attached values** (e.g. `-j1`, `-Ctmp`, `-oquiet`) are applied to mise instead of being swallowed as the task name. That failure mode comes from task positionals allowing hyphen-prefixed values while clap still treats glued short flags as positionals.
> 
> **`escape_task_args`** now discovers value-taking short flags from the CLI/`run` definitions and rewrites matching tokens to **`--long=value`** (including `-C=-dir` when the value itself starts with `-`) before clap runs. Regression tests cover split forms, `=` forms, and full parse into `cd` / task name.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 8146dee0920b1cb9c4221121ece12391559f96a9. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved command-line option handling for short options with attached values, such as `-j1`, `-Ctmp`, and `-oprefix`.
  * Values beginning with a hyphen are now correctly separated and interpreted during task parsing.

### CODE REVIEW

File: src/cli/mod.rs [hunk 1]
```diff
@@ -588,6 +588,21 @@ fn escape_task_args(cmd: &clap::Command, args: &[String]) -> Vec<String> {
...
+                let attached_short_value = arg
+                    .get(..2)
+                    .filter(|_| arg.len() > 2)
+                    .filter(|flag| flags_with_values.iter().any(|f| f == flag));
+
+                if let Some(flag) = attached_short_value {
+                    result.push(flag.to_string());
+                    result.push(arg[flag.len()..].to_string());
+                    i += 1;
+                    continue;
```
[the remark this answers is not in the document]
Comment (jdx) [reply src/cli/mod.rs]: I verified this against the actual parser: after normalization, clap keeps `-dir` bound as the value of `-C` (`cli.cd == PathBuf::from("-dir")`) and still parses `atask` as the task. Commit 0516ae01b adds a parser-level regression test for this exact case, so no behavioral change is needed here.

*AI-assisted — Tool: Codex; model: unavailable; version: unavailable.*

File: src/cli/mod.rs [hunk 2]
```diff
@@ -588,6 +588,21 @@ fn escape_task_args(cmd: &clap::Command, args: &[String]) -> Vec<String> {
...
+                // values when the positional allows hyphens. Split them before
+                // parsing so `-j4`, `-Cdir`, and similar forms remain options.
+                let attached_short_value = arg
+                    .get(..2)
+                    .filter(|_| arg.len() > 2)
+                    .filter(|flag| flags_with_values.iter().any(|f| f == flag));
+
+                if let Some(flag) = attached_short_value {
+                    result.push(flag.to_string());
+                    result.push(arg[flag.len()..].to_string());
```
[the remark this answers is not in the document]
Comment (jdx) [reply src/cli/mod.rs]: Fixed in 0516ae01b. The normalizer now strips one optional `=` delimiter from attached short-option values, and the regression test covers `-o=prefix` alongside the existing attached forms.

*AI-assisted — Tool: Codex; model: unavailable; version: unavailable.*

File: src/cli/mod.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (jdx) [reply src/cli/mod.rs:630]: Addressed in 8146dee09 with a stronger normalization: attached short-option values are rewritten to their long `--name=value` form, avoiding positional ambiguity for ordinary, equals-delimited, and hyphen-prefixed values. Parser-level regressions now cover `-j1`, `-Ctmp`, `-oprefix`, `-C=/tmp`, `-o=prefix`, and `-C-dir`.

*AI-assisted — Tool: Codex; model: unavailable; version: unavailable.*
